### ISSUE TITLE
The legacy of `ErrorCache`: What if there are situations where errors persist?

### ISSUE THREAD

**Issue by Gnuxie** [the-draupnir-project/Draupnir#92]
A remaining unmitigated reason for why ErrorCache was added was because if Mjolnir had been offline and is unable to ban a bunch of new members when it comes back up (receiving lots of join events from sync) it will spam the exact errors for every member event). The management room also reports the errors because Draupnir forcibly verifies permissions at startup and performs a full sync.

This problem really is "find some way to classify whether these errors are the same errors sorts of errors as before, ie is it the same problem and do i already know about it?"
We could be lucky and this is a problem people have to solve all the time and is well documented.

Follow up from https://github.com/Gnuxie/Draupnir/pull/85

**Comment by MTRNord** [comment]
My mind drifted to "throw it into a set and forget" 😅 since a set would just drop it if it already exists. Not sure if that's a good solution though

**Comment by Gnuxie** [comment]
I don't think we really have this issue in v2.0.0? We're much smarter about when we run protections.
